### DISCUSSION TITLE
Split with respect to remote

### DISCUSSION THREAD

**Discussion by lcnbr** [jj-vcs/jj#5124] [Q&A] [answered]
Hello all,

I did not realise that after pushing to a github repo, I had not done `jj new`, thus the local state was the same as on the remote. I then added some more changes but in fact wanted them to be in a new change. Is there a way to tell jj split to split the commit into two, one identical to the remote and then the rest of the changes?

**Comment by scott2000** [comment]
If you run `jj new` now and then run `jj restore --from <branch>@origin --to <branch> --restore-descendants`, I think it should work.

**Reply by lcnbr** [reply]
Nice but this still modifies that commit such that it does not have the same hash as the remote, and thus still needs to be force pushed..

**Reply by scott2000** [reply, marked as answer]
Now that you have the changes in a new commit, you could `jj rebase -s <new_commit> -d <branch>@origin` to rebase it onto the original remote branch commit, and then abandon the old rewritten version.

---

Or instead of my original message, maybe this would've worked better:

```
jj new <branch>@origin
jj restore --from <branch>
jj abandon <branch>
```

Which basically makes a new commit on top of the remote branch, and then resets the commit to the contents from the local branch.

**Reply by arxanas** [reply]
> Now that you have the changes in a new commit, you could jj rebase -s <new_commit> -d <branch>@origin to rebase it onto the original remote branch commit, and then abandon the old rewritten version.

In this case, the desired command might be https://github.com/jj-vcs/jj/issues/1027, which would basically amount to the `new`/`restore`/`abandon` thing more directly, and also preserve the change ID of the local commit (well, I guess I'm not sure if that's desirable for this workflow or not).
